### PR TITLE
use password as seed for signer

### PR DESCRIPTION
**Pull request by fredo**
Every time when the room ensurer starts (once per hour) it uses a new signer and therefore a new user on the other transport servers to login. This is being replaced by using a constant seed phrase to have a only one signer and therefore one user per RSB.

In this PR I used the admin password which is a random string as the seed phrase as it remains the same for the whole lifetime of an RSB provider.

fixes: #228

### CODE REVIEW

File: build/room_ensurer/room_ensurer.py [hunk 1]
```diff
@@ -369,7 +374,7 @@ def _connect(self, server_name: str, server_url: str) -> Optional[Tuple[str, GMa
         password = self._password
 
         if server_name != self._own_server_name:
-            signer = make_signer()
+            signer = LocalSigner(hashlib.sha256(self._password.encode()).digest())
```
Comment (ezdac) [inline build/room_ensurer/room_ensurer.py:377]: Nitpick: I would only initialise this once in the class constructor and reuse the signer here

**Review by ezdac — APPROVED** [review]
Minor performance nitpick. Approved otherwise

**Comment by ezdac** [comment]
Out of curiosity, is the password enforced to be the same throughout the lifetime or is it just expected to not be changed by the user?

**Comment by ulope** [comment]
> Out of curiosity, is the password enforced to be the same throughout the lifetime or is it just expected to not be changed by the user?

It's expected to not change, but if it does that's no big deal either.

The goal of the PR is to not spam the target servers with random users / devices every hour.

File: build/room_ensurer/room_ensurer.py [hunk 2]
```diff
@@ -22,6 +22,8 @@
     - Wait for the room with `#<public_room_alias>:<first_server>` to appear
     - Add server-local alias to the first_server-room
 """
+import hashlib
```
Comment (ulope) [inline build/room_ensurer/room_ensurer.py:25]: This needs to be below the gevent monkeypatch call

**Review by ulope — APPROVED** [review]
LGTM

But see the comment below please.

File: build/room_ensurer/room_ensurer.py (hunk 1 above)
Comment (ulope) [reply build/room_ensurer/room_ensurer.py:377]: True, but the number of servers is quite low, so it shouldn't make a difference in practice

Comment (ezdac) [reply build/room_ensurer/room_ensurer.py:377]: Agreed, also once an hour is not a too frequent interval to make a difference even for high number of servers

**Pull request merged**